diff --git a/win64.mak b/win64.mak
index 5ca1382f..88ea77e0 100644
--- a/win64.mak
+++ b/win64.mak
@@ -2,17 +2,14 @@
 
 MODEL=64
 
-VCDIR=\Program Files (x86)\Microsoft Visual Studio 10.0\VC
-SDKDIR=\Program Files (x86)\Microsoft SDKs\Windows\v7.0A
-
 DMD_DIR=..\dmd
 BUILD=release
 OS=windows
 DMD=$(DMD_DIR)\generated\$(OS)\$(BUILD)\$(MODEL)\dmd
 
-CC="$(VCDIR)\bin\amd64\cl"
-LD="$(VCDIR)\bin\amd64\link"
-AR="$(VCDIR)\bin\amd64\lib"
+CC="$(VCBIN_DIR)\cl"
+LD="$(VCBIN_DIR)\link"
+AR="$(VCBIN_DIR)\lib"
 CP=cp
 
 DOCDIR=doc
@@ -24,8 +21,8 @@ DFLAGS=-m$(MODEL) -conf= -O -release -dip1000 -inline -w -Isrc -Iimport
 UDFLAGS=-m$(MODEL) -conf= -O -release -dip1000 -w -Isrc -Iimport
 DDOCFLAGS=-conf= -c -w -o- -Isrc -Iimport -version=CoreDdoc
 
-#CFLAGS=/O2 /I"$(VCDIR)"\INCLUDE /I"$(SDKDIR)"\Include
-CFLAGS=/Z7 /I"$(VCDIR)"\INCLUDE /I"$(SDKDIR)"\Include
+#CFLAGS=/O2 /I"$(VCINSTALLDIR)"\INCLUDE /I"$(WindowsSdkDir)"\Include
+CFLAGS=/Z7 /I"$(VCINSTALLDIR)"\INCLUDE /I"$(WindowsSdkDir)"\Include
 
 DRUNTIME_BASE=druntime$(MODEL)
 DRUNTIME=lib\$(DRUNTIME_BASE).lib
