diff --git a/win32.mak b/win32.mak
index 0fd01770a..f1f2ab29c 100644
--- a/win32.mak
+++ b/win32.mak
@@ -1,18 +1,16 @@
-MAKE=make
-
 defaulttarget:
 	cd src
-	$(MAKE) -f win32.mak
+	make -f win32.mak
 	cd ..
 
 auto-tester-build:
 	cd src
-	$(MAKE) -f win32.mak auto-tester-build
+	make -f win32.mak auto-tester-build
 	cd ..
 
 auto-tester-test:
 	cd test
-	$(MAKE)
+	gmake -j$(PARALLELISM)
 	cd ..
 	cd samples
 	gmake -f win32.mak DMD=..\src\dmd.exe MODEL=$(MODEL) "LIB=..\..\phobos;$(LIB)" \
